src/document/DocumentAnnotator.ts

```diff
@@ -99,27 +123,25 @@ export default class DocumentAnnotator extends BaseAnnotator {
     };
 
     handleSelectionChange = (): void => {
-        // Clear previous selection immediately
-        this.store.dispatch(setSelectionAction(null));
-        clearTimeout(this.selectionChangeTimer);
+        if (this.isMouseSelecting) {
+            return;
+        }
 
-        this.selectionChangeTimer = window.setTimeout(() => {
-            this.store.dispatch(setSelectionAction(getSelection()));
-        }, TEXT_LAYER_ENHANCEMENT);
+        this.store.dispatch(setSelectionAction(getSelection()));
     };
 
-    debounceHandleSelectionChange = debounce(this.handleSelectionChange, SELECTION_CHANGE_DEBOUNCE, {
-        // The previous selection needs to be cleared immediately when selection changes
-        // Below options make sure the function is triggered on the leading edge of the timeout,
-        // instead of on the trailing edge
-        leading: true,
-        trailing: false,
-    });
+    debounceHandleSelectionChange = debounce(this.handleSelectionChange, SELECTION_CHANGE_DEBOUNCE);
 
     render(): void {
         // Clear previous selection
         this.store.dispatch(setSelectionAction(null));
 
+        if (!this.mouseEventHandlerAdded && this.annotatedEl) {
+            this.annotatedEl.addEventListener('mousedown', this.handleMouseDown);
+            this.annotatedEl.addEventListener('mouseup', this.handleMouseUp);
```

Comment:
What happens if the mouse button is released outside of the annotated element or vice versa? Do things work as desired?

Reply:
If the mouse is released outside of the annotated element or vice versa, the popup will not show, which is the desired behavior confirmed with designer.